vivisect/qt/funcgraph.py

```diff
@@ -455,16 +380,94 @@ def _edgesDone(self, data):
         ''' % addr, self._finishFuncRender)
 
     def _layoutEdges(self, data):
+        '''
+        So...before this, we would only highlight part of the edge lines. That's because
+        The dynadag layout class shoves a bunch of ghost nodes and edges into the graph
+        class member, and each of those end up getting their own html polyline element.
+
+        It'd be a massive change to the Dynadag class to rip those out, and those edges are necessary.
+        So instead, we take care of things here. What the edges/nodes end up looking like in
+        the graph after the dynadag layout is:
+        (VA1, VA2)
+        (VA3, GHOST_NODE1)
+        (GHOST_NODE1, GHOST_NODE2)
+        (GHOST_NODE2, GHOST_NODE3)
+        (GHOST_NODE3, VA3)
+
+        Where a bunch of ghost edges for layout purposes were inserted between the codeblocks for
+        VA3 and VA4. So this function deals with that via creating a bunch of paths, keeping
+        in mind the possibility that an edge line can possibly split for things like switch
+        statements, and links all the points on the edge paths together to make one big
+        polyline.
+
+        This is done in two stages. First loop is to identify the possible series of ghost edges,
+        and only graph the starting point, where we have (Actual_VA1, Ghost_Node). The other
+        cases, like (Ghost_Node, Actual_VA2), (Ghost_Node, Ghost_Node) are ignored, since the
+        second loop picks those up, and the (Actual_VA1, Actual_VA2) is dealt with since that's the
+        base, super easy just do the thing case.
+
+        The second loop just follows all the edge lines and ghost edges to make the one big polyline
+        for each edge so the entire thing can be highlighted via the plineover function we've got
+        stashed in envi/qt/html.py
+        '''
         edgejs = ''
         svgid = 'funcgraph_%.8x' % self.fva
-        for eid, n1, n2, einfo in self.graph.getEdges():
+        graph = self.graph
+        todo = []
+        for eid, n1, n2, einfo in graph.getEdges():
+            src = graph.getNode(n1)
+            dst = graph.getNode(n2)
             points = einfo.get('edge_points')
-            pointstr = ' '.join(['%d,%d' % (x, y) for (x, y) in points])
 
-            edgejs += f'drawSvgLine("{svgid}", "edge_%.8s", "{pointstr}");' % eid
+            # if neither are ghosts, cool. just make the edge
```

Comment:
ZOMG! THIS IS F*ING AMAZING! FuncGraph line highlighting works *FOR THE FIRST TIME SINCE CUTTING TO QT!!!!!!*